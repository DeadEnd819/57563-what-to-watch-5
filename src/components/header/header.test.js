import React from "react";
import renderer from "react-test-renderer";
import Header from "./header";

jest.mock(`../logo/logo.jsx`, () => `Logo`);
jest.mock(`../user-block/user-block.jsx`, () => `UserBlock`);

describe(`Render Header`, () => {
  it(`Should the Header with the user block render correctly`, () => {
    const tree = renderer
      .create(
          <Header className={`movie-card__head`} showUserBlock={true}>
            <React.Fragment/>
          </Header>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should Header render correctly`, () => {
    const tree = renderer
      .create(
          <Header className={`user-page__head`} showUserBlock={false} />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

});
