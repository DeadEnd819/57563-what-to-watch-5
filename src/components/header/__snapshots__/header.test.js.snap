// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Header Should Header render correctly 1`] = `
<header
  className="page-header user-page__head"
>
  <Logo />
</header>
`;

exports[`Render Header Should the Header with the user block render correctly 1`] = `
<header
  className="page-header movie-card__head"
>
  <Logo />
  <UserBlock />
</header>
`;
